src/self-managed-debug/src/system_catalog_dumper.rs

```diff
@@ -206,3 +217,577 @@ pub fn spawn_sql_port_forwarding_process(
         }
     })
 }
+
+#[derive(Debug, Clone)]
+pub enum RelationCategory {
+    /// For relations that belong in the `mz_introspection` schema.
+    /// These relations require a replica name to be specified.
+    Introspection,
+    /// For relations that are retained metric objects that we'd also like to get the SUBSCRIBE output for.
+    Retained,
+    /// Other relations that we want to do a SELECT * FROM on.
+    Basic,
+}
+
+#[derive(Debug, Clone)]
+pub struct Relation {
+    pub name: &'static str,
+    pub category: RelationCategory,
+}
+
+const RELATIONS: &[Relation] = &[
```

Comment:
`static` probably works better here instead of `const`.

When something is `const` the compiler will inline the entire list wherever it's used, so this list will get copied around a bunch. When it's `static` we'll have a single copy in static memory that we pass around references to.